### ISSUE TITLE
Config server not providing profile-specific properties when profile is defined in remote application.yml

### ISSUE THREAD

**Issue by lukasz-kaminski** [spring-cloud/spring-cloud-config#558]
I believe I've found a bug. When profile is defined via `application.yml` located in configuration repository it's not taken into account when providing profile-specific properties. Any other way of setting the profile (e.g. `-Dspring.profiles.active`) works just fine.

### Steps to reproduce:

1. In configuration repository place `myapp.yml` and `myapp-test.yml`. 
Let the `myapp.yml` file contain active profiles setting:

```
spring:
  profiles:
    active: test
```

2. Run MyApp

```
mvn spring-boot:run
```

### Actual result

Only properties from `myapp.yml` are provided.

Logs say:

```
2016-11-21 16:39:55.764  INFO 13486 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Located environment: name=product-service, profiles=[default], label=master, version=cedc977e59902c5ec84661f78acee3104e5701e0
2016-11-21 16:39:55.765  INFO 13486 --- [           main] b.c.PropertySourceBootstrapConfiguration : Located property source: CompositePropertySource [name='configService', propertySources=[MapPropertySource [name='[...]/myapp.yml']]]
2016-11-21 16:39:55.904  INFO 13486 --- [           main] p.e.p.MyApp        : The following profiles are active: test
```

### Expected result:

Properties from `myapp-test.yml` are also provided.

**Comment by chuhuethao** [comment]
I think this issue can be closed as it is similar to this issue? #367

**Comment by lukasz-kaminski** [comment]
Yes, you're right. Closing as a duplicate of #367
